dotCMS/src/main/java/com/dotmarketing/tag/business/TagFactoryImpl.java

```diff
@@ -651,6 +654,40 @@ public List<Tag> getTagsByInodeAndFieldVarName(String inode, String fieldVarName
         return tags;
     }
 
+    @Override
+    public Set<String> getTopTagsBySiteId(final String siteId) throws DotDataException {
+
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(siteId));
+
+        final String selectTopTagsQuery =
+                "select distinct(tagname), count(tinode) from " +
+                        "( " +
+                        "   select lower(tagname) as tagname, tag_inode.inode as tinode " +
+                        "   from  " +
+                        "   tag, tag_inode  " +
+                        "   where  " +
+                        "   tag.tag_id=tag_inode.tag_id  " +
+                        "   and tag.host_id=? " +
+                        "UNION ALL " +
+                        "   select lower(tagname) as tagname, tag_inode.inode as tinode " +
+                        "   from  " +
+                        "   tag, tag_inode  " +
+                        "   where  " +
+                        "   tag.tag_id=tag_inode.tag_id  " +
+                        ") as foo  " +
+                        "group by tagname  " +
+                        "order by count(tinode) desc " +
+                        "limit 1000";
```

Comment:
Shouldn't this limit be customizable via a variable passed down in the method's signature?